test/SlowTests/Issues/RavenDB_22829_DeferrableTimerTests.cs

```diff
@@ -106,7 +106,7 @@ private static async Task ScheduleOnceDeferMany(State state, DeferrableTimeout.P
             
             state.WaitUntilScheduled.Set();
             await Task.Delay(TimeSpan.FromMilliseconds(100)); // simulate some work
-            state.Count++; // this should be thread safe to do
+            Interlocked.Increment(ref state.Count);
```

Comment:
hmmm... given the comment, it might happened that we have a real concurrency issue here?